### PR TITLE
feat: new option `webdriver_proxy` to tunnel requests to the remote Server

### LINKED ISSUE #41: Add ability to connect to selenium hub from inside a proxy
#### Use case
I am using codeception behind a corporate proxy. My selenium grid is outside of the corporate network so I have to use the proxy server to connect to it.
In the suite config yml file you can specify `http_proxy` and `http_proxy_port`, but these are the settings that get injected into the browser itself to connect to the test object url. So we can't take use of them for this use case.

#### Solution
My suggested solution is to add two config keys to the suite config yml: `webdriver_proxy` and `webdriver_proxy_port`. These are passed to the facebook RemoteWebDriver which already supports this kind of proxy settings.

### PR DESCRIPTION
**Pull request by hgsgtk**
Resolves https://github.com/Codeception/module-webdriver/issues/41

## Overview
- When we write configuration `acceptance_suite.yml` like that:

```yml
actor: AcceptanceTester
modules:
    enabled:
        - WebDriver:
              url: https://example.com
              host: selenium-server
              browser: chrome
              webdriver_proxy: log-proxy
              webdriver_proxy_port: 8080
        - \Helper\Acceptance
step_decorators: ~       
```

- We could pass `httpProxy` and `httpProxyPort` options to php-webdriver correctly.

### CODE REVIEW

**Comment by hgsgtk** [comment]
CI is failed, but it seemed to be an error already before.

https://semaphoreci.com/codeception/module-webdriver/branches/master/builds/42

Does anyone know the cause of it?

File: src/Codeception/Module/WebDriver.php [hunk 1]
```diff
@@ -404,10 +408,8 @@ class WebDriver extends CodeceptionModule implements
     protected $test;
     protected $sessions = [];
     protected $sessionSnapshots = [];
-    protected $httpProxy;
```
Comment (hgsgtk) [inline src/Codeception/Module/WebDriver.php:407]: These properties were created at the following commit.

https://git.hrzg.de/t.munk/Codeception/commit/43f6e26d5f010dd3aa390252e3d7a10efd733c48?w=1

However, it's not used from anywhere.

Current `httpProxy` and `httpProxyPort` option are passed to capabilities (to put an proxy between webdriver server and SUT).

Comment (hgsgtk) [reply src/Codeception/Module/WebDriver.php:407]: sslProxy and sslProxyPort is also.

File: src/Codeception/Module/WebDriver.php [hunk 2]
```diff
@@ -1596,8 +1600,8 @@ public function _initializeSession()
                 $this->capabilities,
                 $this->connectionTimeoutInMs,
                 $this->requestTimeoutInMs,
-                $this->httpProxy,
```
Comment (hgsgtk) [inline src/Codeception/Module/WebDriver.php:1599]: Now, it passes null always because httpProxy is not set from anywhere.

**Comment by Naktibalda** [comment]
> 
> 
> CI is failed, but it seemed to be an error already before.
> 
> https://semaphoreci.com/codeception/module-webdriver/branches/master/builds/42
> 
> Does anyone know the cause of it?

It is alright, only `Chrome Headless` has been passing for a long time.

**Comment by hgsgtk** [comment]
I got it, thanks!

image

I undestood it means a check on this PR branch is passed.

File: src/Codeception/Module/WebDriver.php (hunk 1 above)
Comment (Naktibalda) [reply src/Codeception/Module/WebDriver.php:407]: Good point, it seems that these properties have never been used.

**Comment by DavertMik** [comment]
> It is alright, only Chrome Headless has been passing for a long time.

We need to move all things to GitHub Actions

**Comment by Naktibalda** [comment]
Released as 1.3.0

**Comment by hgsgtk** [comment]
Great, super thanks!!

**Pull request merged**